### PR TITLE
Change focus after saving reference and new sname

### PR DESCRIPTION
**Pull request by tonimobin**
The focus now changes accordingly: 

- Save reference -> Add new structured name
- Add new structured name -> Name
- Save (Sname) -> Add new structured name

Please note that having your console open while testing the focus changes might cause issues (the console might 'hijack' the focus and thus you won't see it). It is also recommended to do the testing with your keyboard as clicking might make the focus disappear. Also, it could be a good idea to add more of these focus changes. For example: 

- Edit (reference) -> first_author
- Manual entry -> first_author
- Make new doi search -> doi
- get -> first_author

But these have not been made yet, as they were not requested. However, if you find ones that speed up the process of data entry - feel free to propose them in Slack/Scrum etc.

### CODE REVIEW

**Comment by tonimobin** [comment]
setTimeout might cause issues in the future so it needs to be fixed. PR has now been changed to draft state.

**Comment by tonimobin** [comment]
Following focus changes exist now:
Wizard loaded for the first time -> doi
doi get -> first author
save reference -> add new structured name
add new structured name -> name
save duplicate sname via ok -> add new structured name
cancel duplicate sname via cancel -> add new structured name
manual entry -> first author
edit -> first author 
new doi search -> doi

Timeouts raised from 20ms to 200ms, but should still be converted to utilize some sort of a ref.

**Comment by tonimobin** [comment]
Removed focus changes that utilize timeouts. Added new focus changes.

The following focus changes now exist:
wizard loaded for the first time -> doi
doi get -> first author
save reference -> add new structured name
add new structured name -> name
save duplicate sname via ok -> add new structured name
cancel duplicate sname via cancel -> add new structured name
manual entry -> first author
edit -> first author
new doi search -> doi
save reference via edit form -> add new structured name
delete created sname -> add new structured name
delete created sname, when another create sname is open -> add new structured name
save reference / save reference via edit, when create sname is open -> name
delete existing sname -> select existing name

File: app/frontend/src/App.js [hunk 1]
```diff
@@ -118,20 +126,26 @@ const App = () => {
 				<Notification notification={nameNotification}/>
 				<Notification notification={locationNotification}/>
 				{state.sname.map(sname => (
-					<Sname {...{ key: sname.id, sname }} canDeleteNotify= {canDeleteNotify} nameNotify={nameNotify} locationNotify={locationNotify}/>
+					<Sname {...{ key: sname.id, sname }} canDeleteNotify= {canDeleteNotify} nameNotify={nameNotify} locationNotify={locationNotify} deleteCreatedSname setDeleteCreatedSname={setDeleteCreatedSname}/>
```
Comment (qubelka) [inline app/frontend/src/App.js:129]: deleteCreatedSname ei taida olla käytössä Sname komponentissa, joten sen voi poistaa propseista

File: app/frontend/src/components/SnameForm.js [hunk 2]
```diff
@@ -109,8 +127,44 @@ export const SnameForm = ({
...
+	useEffect(() => {
+		if(newSnameButtonIsDisabled){
+			nameRef.current.focus()
+		}
+		if (displayRefForm === 'none' && newSnameButtonIsDisabled) {
+			setFocusOnSnameButton()
+		}
+		if (displayRefForm === 'none' && !newSnameButtonIsDisabled) {
+			setFocusOnSnameButton()
+		}
```
Comment (qubelka) [inline app/frontend/src/components/SnameForm.js:143]: Nämä kaksi voi yhdistää: 
`	
if (displayRefForm === 'none') {
  setFocusOnSnameButton()
}
`

**Review by qubelka — APPROVED** [review]
Hienoa! 👍

**Pull request merged**